Cython/Utility/ModuleSetupCode.c

```diff
@@ -307,6 +307,12 @@
   #define __has_cpp_attribute(x) 0
 #endif
 
+/* This macro can be used to suppress warnings about constant conditional expressions */
+#ifndef __PYX_CONST_CONDITION
+  static int __Pyx_const_condition(int b) { return b; }
```

Comment:
Does this indirection prevent any compilers from actually optimizing the relevant branches away (which is quite important)?

Reply:
As long as optimizations are activated any compiler I know will inline this call. Afterwards the same optimizations are performed as without this call. Obviously you can get the compiler to emit worse code e.g. by disabling inlining `/O2 /Ob0`, which I do not think is a concern.